internal/validator/handler.go

```diff
@@ -42,18 +44,45 @@ type handler struct {
 	producer *kafka.Producer
 	schema   *jsonschema.Schema
 	errors   chan error
+	msgChan  chan *kafka.Message
+	vChan    chan messageInfo
 }
 
 func (this *handler) onMessage(ctx context.Context, msg *kafka.Message) {
-	request := &messageModel.IngressValidationRequest{}
-	err := json.Unmarshal(msg.Value, request)
+	wg.Add(1)
 
-	if err != nil {
-		instrumentation.UnmarshallingError(ctx, err)
-		return
-	}
+	go func() {
+		defer wg.Done()
+
+		for {
+			if len(this.msgChan) >= cap(this.msgChan) {
+				continue
+			}
+			this.msgChan <- msg
+			break
+		}
+	}()
+	wg.Wait()
```

Comment:
Unless I missed something this looks equivalent to simply doing
```
func (this *handler) onMessage(ctx context.Context, msg *kafka.Message) {
    this.msgChan <- msg
}
```

Reply:
Since this is a buffered channel, I think that will end up resulting in deadlock issues when the `msgChan` buffer is full. My intention here was to wait until a place becomes available on the channel before attempting to add the message to it.

Reply:
Can you elaborate on what sequence of events would lead into the deadlock?

Reply:
I am currently limiting the maximum number of goroutines that gets spun in `fetchMessage`. We can circle back on this conversation once that work is complete because the new changes in `fetchMessage` might have an affect on this.